zerver/lib/user_topics.py

```diff
@@ -352,3 +352,31 @@ def get_users_with_user_topic_visibility_policy(
     return UserTopic.objects.filter(
         stream_id=stream_id, topic_name__iexact=topic_name
     ).select_related("user_profile", "user_profile__realm")
+
+
+def get_moved_messages_senders_to_follow_target_topic(
+    orig_stream_id: int,
+    orig_topic: str,
+    target_stream_id: int,
+    target_topic: str,
+    sender_ids: list[int],
+) -> QuerySet[UserTopic]:
+    return (
+        UserTopic.objects.filter(
+            user_profile__id__in=sender_ids,
```

Comment:
Nit: to reduce an unnecessary join
```suggestion
            user_profile_id__in=sender_ids,
```